pkg/sdk/logging/api/v1beta1/loggingroute_types.go

```diff
@@ -0,0 +1,66 @@
+// Copyright © 2023 Kube logging authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type LoggingRouteSpec struct {
+	// Source identifies the logging that this policy applies to
+	Source string `json:"source"`
+
+	// Targets refers to the list of logging resources specified by a label selector to forward logs to
+	// Filtering of namespaces will happen based on the watchNamespaces and watchNamespaceSelector fields of the target logging resource
+	Targets metav1.LabelSelector `json:"targets"`
+}
+
+type LoggingRouteStatus struct {
+	// Enumerate all loggings with all the destination namespaces expanded
+	Tenants []Tenant `json:"tenants,omitempty"`
+
+	// Enumerate problems that prohibits this policy to take effect
```

Comment:
[nitpicking]
The comment is slightly _wrong_, because in the following case the policy will take effect, but the problems field will be filled:
```
problems = append(problems, fmt.Sprintf("tenant %s receives logs from ALL namespaces", t.Name))
```

Reply:
I'm thinking of adding a separate warning field instead

Reply:
https://github.com/kube-logging/logging-operator/pull/1476/commits/5dc4398c0ff027a6f673fb77770725db8b6c8397